spock/utils.py

```diff
@@ -59,45 +59,44 @@ class BufferUnderflowException(Exception):
 
 
 class BoundBuffer(object):
-    backup = b''
+    buff = b''
+    cursor = 0
 
-    def __init__(self, *args):
-        self.count = 0
-        self.buff = (args[0] if args else b'')
+    def __init__(self, data=b""):
+        self.write(data)
 
-    def recv(self, bytes):
-        if len(self.buff) < bytes:
+    def read(self, length):
+        if length > len(self):
             raise BufferUnderflowException()
-        self.count += bytes
-        o, self.buff = self.buff[:bytes], self.buff[bytes:]
-        return o
 
-    def append(self, bytes):
-        self.buff += bytes
+        out = self.buff[self.cursor:self.cursor+length]
+        self.cursor += length
+        return out
+
+    def write(self, data):
+        self.buff += data
 
     def flush(self):
-        out = self.buff
-        self.buff = b''
-        self.save()
```

Comment:
Note that flush() no longer calls save(), because I don't think it should - it should just be a read() of the whole buffer. Any thoughts?